nightfall-optimist/src/services/database.mjs

```diff
@@ -27,13 +27,16 @@ export async function saveCommit(commitHash, txDataToSign) {
   return db.collection(COMMIT_COLLECTION).insertOne({ commitHash, txDataToSign });
 }
 /**
-Function to retrieve a commit, by commitHash
+Function to retrieve a commit, by commitHash, it also returns the 'retrieved'
+which will be true if the commitment hash has already been retrieved
 */
 export async function getCommit(commitHash) {
   const connection = await mongo.connection(MONGO_URL);
   const db = connection.db(OPTIMIST_DB);
   const query = { commitHash };
-  return db.collection(COMMIT_COLLECTION).findOne(query);
+  const commit = await db.collection(COMMIT_COLLECTION).findOne(query);
+  if (commit) db.collection(COMMIT_COLLECTION).updateOne(query, { $set: { retrieved: true } });
```

Comment:
I believe this will need to be `awaited` or we will end up returning before the update is run and the commit marked as `retrieved`?

Reply:
Good shout.

Reply:
fixed